chain/pool/txpool.go

```diff
@@ -138,50 +110,44 @@ func (tp *TxnPool) processTx(txn *transaction.Transaction) error {
 	case pb.NanoPayType:
 		tp.NanoPayTxs.Store(txn.Hash(), txn)
 	default:
-		isOrphan := true
-		replace := false
-		if !list.Empty() {
-			//replace old tx that has same nonce.
-			if _, err := list.Get(txn.UnsignedTx.Nonce); err == nil {
-				log.Warning("replace old tx")
-				//TODO need more fee
-				isOrphan = false
-				replace = true
-			} else if list.Full() {
-				return errors.New("txpool is full")
-			} else if preNonce, _ := list.GetLatestNonce(); txn.UnsignedTx.Nonce == preNonce+1 {
-				isOrphan = false
+		if _, err := list.Get(txn.UnsignedTx.Nonce); err == nil {
+			log.Warning("replace old tx")
+			if err := tp.blockValidationState.VerifyTransactionWithBlock(txn, nil); err != nil {
```

Comment:
Should we replace the old txn in blockValidationState instead of trying to add it to blockValidationState? also @trueinsider

Reply:
yeah, we should clean old tx before adding new one

Reply:
We should also make sure that blockValidationState is not modified if processTx returns error

Reply:
we can divide verify bvs into verify and append

Reply:
or simply do clean in case of error

Reply:
Let's do this in a separate PR